src/crypto.c

```diff
@@ -44,39 +45,40 @@ int crypto_derive_private_key(cx_ecfp_private_key_t *private_key,
                                      private_key);
         }
         CATCH_OTHER(e) {
-            THROW(e);
+            error = e;
         }
         FINALLY {
             explicit_bzero(&raw_private_key, sizeof(raw_private_key));
         }
     }
     END_TRY;
 
-    return 0;
+    return error;
 }
 
-int crypto_init_public_key(cx_ecfp_private_key_t *private_key,
-                           cx_ecfp_public_key_t *public_key,
-                           uint8_t raw_public_key[static 64]) {
+void crypto_init_public_key(cx_ecfp_private_key_t *private_key,
+                            cx_ecfp_public_key_t *public_key,
+                            uint8_t raw_public_key[static 64]) {
     // generate corresponding public key
     cx_ecfp_generate_pair(CX_CURVE_256K1, public_key, private_key, 1);
 
     memmove(raw_public_key, public_key->W + 1, 64);
-
-    return 0;
 }
 
-int crypto_sign_message() {
+int crypto_sign_message(void) {
     cx_ecfp_private_key_t private_key = {0};
     uint8_t chain_code[32] = {0};
     uint32_t info = 0;
     int sig_len = 0;
 
     // derive private key according to BIP32 path
-    crypto_derive_private_key(&private_key,
-                              chain_code,
-                              G_context.bip32_path,
-                              G_context.bip32_path_len);
+    int error = crypto_derive_private_key(&private_key,
+                                          chain_code,
+                                          G_context.bip32_path,
+                                          G_context.bip32_path_len);
+    if (error != 0) {
+        goto end;
```

Comment:
The rest of the boilerplate code seems happy with functions with multiple return statements, so as there's not much to gain with a `goto` here (no common cleaning steps or whatnot), a `return` should be plenty.

Reply:
Fixed :+1: